xml/value.go

```diff
@@ -0,0 +1,257 @@
+package xml
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"math"
+	"math/big"
+	"strconv"
+)
+
+// Value represents an XML Value type
+// XML Value types: Object, Array, Map, String, Number, Boolean, and Null.
+type Value struct {
+	w       *bytes.Buffer
+	scratch *[]byte
+
+	openTagWriterFn  func()
+	closeTagWriterFn func()
```

Comment:
nit, its probably better for these functions to take a writer, and return an error as well. This way the writer can exist only a consistent top down flow passed in instead of existing in various closures in the xml node stack.

```go
openTagWriterFn func(w *bytes.Buffer) error
```